src/services/bill_manager/bulk_invoice.rs

```diff
@@ -0,0 +1,72 @@
+use crate::client::{Mpesa, MpesaResult};
+use crate::constants::Invoice;
+use crate::environment::ApiEnvironment;
+use crate::errors::MpesaError;
+use serde::Deserialize;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct BulkInvoiceResponse {
+    #[serde(rename(deserialize = "rescode"))]
+    pub res_code: String,
+    #[serde(rename(deserialize = "resmsg"))]
+    pub res_msg: String,
+    #[serde(rename(deserialize = "Status_Message"))]
+    pub status_message: String,
+}
+
+#[derive(Debug)]
+pub struct BulkInvoiceBuilder<'mpesa, Env: ApiEnvironment> {
+    client: &'mpesa Mpesa<Env>,
+    invoices: Option<Vec<Invoice<'mpesa>>>,
+}
+
+impl<'mpesa, Env: ApiEnvironment> BulkInvoiceBuilder<'mpesa, Env> {
+    /// Creates a new Bill Manager Bulk Invoice builder
+    pub fn new(client: &'mpesa Mpesa<Env>) -> BulkInvoiceBuilder<'mpesa, Env> {
+        BulkInvoiceBuilder {
+            client,
+            invoices: None,
+        }
+    }
+
+    /// Adds `invoices`
+    pub fn invoices(mut self, invoices: Vec<Invoice<'mpesa>>) -> BulkInvoiceBuilder<'mpesa, Env> {
+        self.invoices = Some(invoices);
+        self
+    }
```

Comment:
Design question/ suggestion

Rather than initializing the `BulkInvoiceBuilder` invoices value as `None`, wouldn't it be easier to initialize it as an empty `Vec`.

Such that `invoices` would be a method that appends to this `Vec`, with the ability to chain the `invoices` calls. The method would probably need to be renamed to `add_invoice` or simply `invoice` and would allow the user to append a single invoice to the existing `Vec`

This being said, I still see the need to have a method to add many invoices in bulk with the option to add a single one, WDYT?

Reply:
Ahh, funny thing is I had initially implemented it that way haha. Yeah I'll add two separate methods. I'd go for `invoice()` and `invoices()` instead of `add_*()`, it's more consistent that way when looking at other modules as well.